### PR TITLE
ATLEDGE-184 SoftwareSerial Library

### PR DESCRIPTION
**Pull request by bigdinotech**
Port of SoftwareSerial Library for Arduino101
Tx works up to 384000 bps
Rx works up to 57600 bps

### CODE REVIEW

**Comment by bigdinotech** [comment]
@kmsywula @SidLeung Ready for code review

File: libraries/SoftwareSerial/SoftwareSerial.cpp [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+        if(i == 8)
+        {
+          delayTicks(firstIntraBitDelay);
+        }
+        else
+        {
+          delayTicks(rxIntraBitDelay);
+        }
+      }
+      d >>= 1;
```
Comment (kmsywula) [inline libraries/SoftwareSerial/SoftwareSerial.cpp]: That would be nice to have here some comment explaining what happens here.

File: libraries/SoftwareSerial/SoftwareSerial.cpp [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+      if (digitalRead(rxPin))
+        d |= 0x80;
```
Comment (kmsywula) [inline libraries/SoftwareSerial/SoftwareSerial.cpp]: That would be nice to have here some comment explaining what happens here.

File: libraries/SoftwareSerial/examples/TwoPortReceive/TwoPortReceive.ino [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  Serial.println("Data from port two:");
+  while (portTwo.available() > 0) {
+    char inByte = portTwo.read();
+    Serial.write(inByte);
+  }
+
+  // blank line to separate data from the two ports:
+  Serial.println();
+}
+
```
Comment (kmsywula) [inline libraries/SoftwareSerial/examples/TwoPortReceive/TwoPortReceive.ino]: Remove additional new lines here.

**Comment by kmsywula** [comment]
@calvinatintel in my opinion it's good enough to be merged. 
@bigdinotech please address remaining comments in next merge

File: libraries/SoftwareSerial/examples/TwoPortReceive/TwoPortReceive.ino (hunk 3 above)
Comment (bigdinotech) [inline libraries/SoftwareSerial/examples/TwoPortReceive/TwoPortReceive.ino]: This example is from the original SoftwareSerial library from Arduino Uno.
I can make the changes still though.

**Comment by bigdinotech** [comment]
@calvinatintel I have addressed the cosmetic issues. Ready for cherry-pick

**Pull request merged**